src/components/Commits/commit-status.ts

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { PipelineRunLabel } from '../../consts/pipelinerun';
+import { PipelineRunGroupVersionKind } from '../../models';
+import { pipelineRunFilterReducer } from '../../shared';
+import { PipelineRunKind } from '../../types';
+import { statuses } from '../../utils/commits-utils';
+import { useNamespace } from '../../utils/namespace-context-utils';
+
+export const useCommitStatus = (
+  application: string,
+  commit: string,
+): [string, boolean, unknown] => {
+  const namespace = useNamespace();
+  const [pipelineruns, loaded, loadErr] = useK8sWatchResource<PipelineRunKind[]>({
+    groupVersionKind: PipelineRunGroupVersionKind,
+    isList: true,
+    namespace,
+  });
+
+  const plrsForCommit = React.useMemo(
+    () =>
+      pipelineruns
+        ?.filter(
+          (plr) =>
+            plr.metadata.labels[PipelineRunLabel.APPLICATION] === application &&
+            plr.metadata.labels[PipelineRunLabel.COMMIT_LABEL] === commit,
+        )
+        .sort((a, b) => parseInt(a.status?.startTime, 10) - parseInt(b.status?.startTime, 10)),
+    [application, commit, pipelineruns],
+  );
+
+  const commitStatus = React.useMemo(() => {
+    if (!loaded || loadErr) {
+      return '-';
+    }
+
+    let stat = '-';
+    plrsForCommit.some((plr) => {
+      const plrStatus = pipelineRunFilterReducer(plr);
+      if (statuses.includes(plrStatus)) {
+        stat = plrStatus;
+        return true;
+      }
+    }, {});
```

Comment:
This logic will return first matched pipelinerun's status against statuses array rather it should be the latest pipelinerun (with commit sha label) status.

eg: If i retrigger the pipelinerun, it still shows the old pipelinerun's status (succeeded)
<img width="1787" alt="Screenshot 2023-01-04 at 1 09 50 PM" src="https://user-images.githubusercontent.com/9964343/210510220-8e511eef-e1db-428e-b677-a0e68599ca04.png">



```suggestion
    const plrStatus = pipelineRunFilterReducer(plrsForCommit[plrsForCommit.length - 1]);
    if (statuses.includes(plrStatus)) {
      stat = plrStatus;
      return stat;
    }
```